sys/include/net/gnrc/dns.h

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright (C) 2015 Nick van IJzendoorn <nijzendoorn@engineering-spirit.nl>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for
+ * more details.
+ */
+
+/**
+ * @defgroup    net_dns DNS
+ * @ingroup     net
+ * @brief       Provides DNS header and helper functions
+ * @{
+ *
+ * @file
+ * @brief       DNS client implementation
+ *
+ * The DNS module get's it's DNS server configuration from the NDP
+ * Router Advertisement message. The implementation loosely follows the
+ * following RFC's:
+ *
+ *  - https://www.ietf.org/rfc/rfc1035.txt
+ *     (RFC1035 Domain Names - Implementation and Specifications)
+ *       - Only implemented A, CNAME, TXT records querying
+ *       - No cache support
+ *  - https://www.rfc-editor.org/rfc/rfc2782.txt
+ *     (RFC2782 A DNS RR for specifying the location of services)
+ *       - Fully
+ *  - https://www.ietf.org/rfc/rfc3596.txt
+ *     (RFC3596 DNS Extensions to Support IPv6)
+ *       - Fully
+ *  - https://tools.ietf.org/html/rfc6106
+ *     (RFC6106 IPv6 Router Advertisement Options for DNS Configuration)
+ *       - Only RDNSS, no DNSSL
+ *
+ * @author      Nick van IJzendoorn <nijzendoorn@engineering-spirit.nl>
+ */
+
+#ifndef GNRC_DNS_H_
+#define GNRC_DNS_H_
+
+#include <stdint.h>
+
+#include "net/gnrc/ipv6/netif.h"
+#include "net/ipv6/addr.h"
+#include "net/ipv4/addr.h"
+#include "byteorder.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @def GNRC_DNS_SERVERS_SUPPORT
+ *
+ * @brief   Number of DNS servers that are supported per interface
+ */
+#ifndef GNRC_DNS_SERVERS_SUPPORT
+#define GNRC_DNS_SERVERS_SUPPORT             (3)
+#endif
+
+/**
+ * @def GNRC_DNS_MSG_QUEUE_SIZE
+ *
+ * @brief   Default message queue size for the DNS thread
+ */
+#ifndef GNRC_DNS_MSG_QUEUE_SIZE
+#define GNRC_DNS_MSG_QUEUE_SIZE              (8U)
+#endif
+
+/**
+ * @def GNRC_DNS_PRIO
+ *
+ * @brief   Priority of the DNS thread
+ */
+#ifndef GNRC_DNS_PRIO
+#define GNRC_DNS_PRIO                        (THREAD_PRIORITY_MAIN - 2)
+#endif
+
+/**
+ * @def GNRC_DNS_STACK_SIZE
+ *
+ * @brief   Default stack size to use for the DNS thread
+ */
+#ifndef GNRC_DNS_STACK_SIZE
+#define GNRC_DNS_STACK_SIZE                  (2048)
```

Comment:
Wouldn't it be preferable to use a value based on a platform dependent value here like `THREAD_STACKSIZE_DEFAULT + X` as we do for the other GNRC stack sizes?